.github/workflows/lint-test-sdk.yml

```diff
@@ -4,11 +4,37 @@ on:
     paths:
       - '**/*'
 
+  workflow_dispatch:
+
+  workflow_call:
+    inputs:
+      test_data_branch:
+        type: string
+        description: The branch in sdk-test-data to target for testcase files
+        required: false
+        default: main
+      sdk_branch:
+        type: string
+        description: The branch of the SDK to test
+        required: false
+
+env:
+  SDK_BRANCH_NAME: ${{ inputs.sdk_branch  || github.head_ref || github.ref_name || 'main' }}
```

Comment:
Do we want a fallback to `main` here? I would think merges to main would be covered by `github.ref_name`, wouldn't it? If that's the case, it might make sense to throw an error if `inputs.sdk_branch  || github.head_ref || github.ref_name` result in an empty `$SDK_BRANCH_NAME` since that would indicate something was wrong with the configuration.

Reply:
You are right that the `main` fallback here is superfluous. Looking at _Actions/checkout_ readme and code, the ref will be computed as the triggering branch or the default branch if `ref` is omitted or is a blank string.

The checkout action will throw an error when the refspec doesn't exist, stopping the workflow and displays an error message indicating as much. I'm not sure throwing our own error here would add much, to be honest, but I am trying to get these all hooked up so I can move on to the SDK packaging and integration testing so I may be biased.

Reply:
I don't mean we need to create our own error. I just meant that it might give a false sense of security if we fallback to main when it might indicate a larger problem with our configuration. In any case, it's not a blocker; just something I wanted to point out as an improvement opportunity in a subsequent iteration.

Reply:
> The checkout action will throw an error when the refspec doesn't exist

Ah, right. This is a good callout. Okay, cool.